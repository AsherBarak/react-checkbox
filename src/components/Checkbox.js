import React, { Component } from 'react';
import './Checkbox.css';

class Checkbox extends Component {
    constructor(props) {
        super(props);
        this.state={value:false};
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick() {
        if (this.props.onChange) this.props.onChange(!this.state.value);
        this.setState((prevState, props)=>({
            value: !prevState.value, 
        }));
    }

    render() {
        let icon= this.state.value
            ? <span> &#9745;</span>
            :<span> &#x2610;</span>

            let cssClass= (this.props.className)
            ?this.props.className
            :'';

        return (
            <div className={cssClass}>
                <span className='icon' onClick={this.handleClick}>{icon}</span> {this.props.children}
            </div>
        );
    }
}

export default Checkbox;