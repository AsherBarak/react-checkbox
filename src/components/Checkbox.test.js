import React from 'react';
import ReactDOM from 'react-dom';
import Checkbox from './Checkbox'
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Checkbox />, div);
  });

// Snapshot
it('Renders consistent tree', () => {
    const component = renderer.create(
      <Checkbox />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });