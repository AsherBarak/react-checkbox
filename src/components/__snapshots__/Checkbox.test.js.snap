// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders consistent tree 1`] = `
<div
  className=""
>
  <span
    className="icon"
    onClick={[Function]}
  >
    <span>
       ‚òê
    </span>
  </span>
   
</div>
`;
